packages/machine/src/protocol/install-virtual-app.ts

```diff
@@ -145,39 +167,56 @@ function proposeStateTransition3(message: ProtocolMessage, context: Context) {
     0,
     defaultTimeout
   );
+
   context.targetVirtualAppInstance = targetAppInstance;
 
-  const rightEthVirtualAppAgreementInstance = new ETHVirtualAppAgreementInstance(
-    context.stateChannelsMap.get(multisig2Address)!.multisigAddress,
+  const channelWithIntermediary = getChannelFromCounterparty(
+    context.stateChannelsMap,
+    initiatingAddress,
+    intermediaryAddress
+  );
+
+  if (!channelWithIntermediary) {
+    throw Error(
+      "Cannot run InstallVirtualAppProtocol without existing channel with intermediary"
+    );
+  }
+
+  const leftETHVirtualAppAgreementInstance = new ETHVirtualAppAgreementInstance(
+    channelWithIntermediary.multisigAddress,
     {
-      assetType: 0,
-      limit: respondingBalanceDecrement,
+      assetType: AssetType.ETH,
+      limit: bigNumberify(initiatingBalanceDecrement).add(
+        respondingBalanceDecrement
+      ),
       token: AddressZero
     },
-    context.stateChannelsMap.get(multisig2Address)!.numInstalledApps + 1,
-    context.stateChannelsMap.get(multisig2Address)!.rootNonceValue,
+    channelWithIntermediary.numInstalledApps + 1,
+    channelWithIntermediary.rootNonceValue,
     100,
-    initiatingBalanceDecrement.add(respondingBalanceDecrement).toNumber()
+    bigNumberify(initiatingBalanceDecrement)
```

Comment:
isn't `initiatingBalanceDecrement` already a `BigNumber`?

Reply:
There's a problem where the value submitted to this has not been deserialized